sysl2/sysl/importer/xsd.go

```diff
@@ -0,0 +1,224 @@
+package importer
+
+import (
+	"bytes"
+	"encoding/xml"
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+
+	"aqwari.net/xml/xsd"
+	"github.com/sirupsen/logrus"
+)
+
+func LoadXSDText(args OutputData, text string, logger *logrus.Logger) (out string, err error) {
+	xsd.StandardSchema = [][]byte{} // Ignore all the standard schemas,
+	specs, err := xsd.Parse([]byte(text))
+	if err != nil {
+		return "", err
+	}
+
+	types := TypeList{}
+	for _, schema := range specs {
+		schemaTypes := loadSchemaTypes(schema, logger)
+		types.Add(schemaTypes.Items()...)
+	}
+	types.Sort()
+
+	info := SyslInfo{
+		OutputData:  args,
+		Description: "",
+		Title:       "",
+	}
+
+	result := &bytes.Buffer{}
+	w := newWriter(result, logger)
+	if err := w.Write(info, types, ""); err != nil {
+		return "", err
+	}
+
+	return result.String(), nil
+}
+
+func makeNamespacedType(name xml.Name, target Type) Type {
+	return &ImportedBuiltInAlias{
+		name:   fmt.Sprintf("%s:%s", name.Space, name.Local),
+		Target: target,
+	}
+}
+
+func loadSchemaTypes(schema xsd.Schema, logger *logrus.Logger) TypeList {
+	types := TypeList{}
+
+	var xsdToSyslMappings = map[string]string{
+		"NMTOKEN":      StringTypeName,
+		"integer":      "int",
+		"time":         StringTypeName,
+		"boolean":      "bool",
+		StringTypeName: StringTypeName,
+		"date":         "date",
+	}
+	for swaggerName, syslName := range xsdToSyslMappings {
+		from := xml.Name{Space: "http://www.w3.org/2001/XMLSchema", Local: swaggerName}
+		to := &SyslBuiltIn{syslName}
+		types.Add(makeNamespacedType(from, to))
+	}
+
+	keys := make([]xml.Name, 0, len(schema.Types))
+	for key := range schema.Types {
+		keys = append(keys, key)
+	}
+	sort.Slice(keys, func(i, j int) bool {
+		return strings.Compare(
+			fmt.Sprintf("%s:%s", keys[i].Space, keys[i].Local),
+			fmt.Sprintf("%s:%s", keys[j].Space, keys[j].Local)) < 0
+	})
+
+	for _, name := range keys {
+		data := schema.Types[name]
+		logger.Infof("name: %+v\n", name)
```

Comment:
Info is too verbose for such logging, please use debug.

Reply:
removed